From b903b967da8f322503619861543d7b743d31bd97 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Miro=20Hron=C4=8Dok?= <miro@hroncok.cz>
Date: Wed, 9 Mar 2022 13:37:49 +0100
Subject: [PATCH] Use ${CMAKE_COMMAND} when running cmake from within cmake

This way, if the command is different (e.g. /usr/bin/cmake3 on EPEL 7),
it still works.
---
 CMakeLists.txt | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d70a030d..563593d4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -88,21 +88,21 @@ add_custom_command(
   WORKING_DIRECTORY
     ${CMAKE_CURRENT_SOURCE_DIR}/AutomaticComponentToolkit
   COMMAND
-    cmake -E make_directory "${CMAKE_CURRENT_SOURCE_DIR_AUTOGENERATED}"
+    ${CMAKE_COMMAND} -E make_directory "${CMAKE_CURRENT_SOURCE_DIR_AUTOGENERATED}"
   COMMAND
-    cmake -E touch "${CMAKE_CURRENT_BINARY_DIR_AUTOGENERATED}/LIB3MFACTGENERATIONSTAMP"
+    ${CMAKE_COMMAND} -E touch "${CMAKE_CURRENT_BINARY_DIR_AUTOGENERATED}/LIB3MFACTGENERATIONSTAMP"
   COMMAND
     ${CMAKE_CURRENT_SOURCE_DIR}/AutomaticComponentToolkit/bin/act.${ACT_COMMANDENDING} ${CMAKE_CURRENT_SOURCE_DIR}/AutomaticComponentToolkit/lib3mf.xml
   COMMAND
-    cmake -E remove_directory "${CMAKE_CURRENT_SOURCE_DIR_AUTOGENERATED}/Source"
+    ${CMAKE_COMMAND} -E remove_directory "${CMAKE_CURRENT_SOURCE_DIR_AUTOGENERATED}/Source"
   COMMAND
-    cmake -E remove_directory "${CMAKE_CURRENT_SOURCE_DIR_AUTOGENERATED}/Bindings"
+    ${CMAKE_COMMAND} -E remove_directory "${CMAKE_CURRENT_SOURCE_DIR_AUTOGENERATED}/Bindings"
   COMMAND
-    cmake -E copy_directory "${CMAKE_TEMP_HEADER_FOLDER}/Bindings" ${CMAKE_CURRENT_SOURCE_DIR_AUTOGENERATED}/Bindings
+    ${CMAKE_COMMAND} -E copy_directory "${CMAKE_TEMP_HEADER_FOLDER}/Bindings" ${CMAKE_CURRENT_SOURCE_DIR_AUTOGENERATED}/Bindings
   COMMAND
-    cmake -E copy_directory "${CMAKE_TEMP_HEADER_FOLDER}/Implementations/Cpp/Interfaces" "${CMAKE_CURRENT_SOURCE_DIR_AUTOGENERATED}/Source/"
+    ${CMAKE_COMMAND} -E copy_directory "${CMAKE_TEMP_HEADER_FOLDER}/Implementations/Cpp/Interfaces" "${CMAKE_CURRENT_SOURCE_DIR_AUTOGENERATED}/Source/"
   COMMAND
-    cmake -E remove_directory "${CMAKE_TEMP_HEADER_FOLDER}"
+    ${CMAKE_COMMAND} -E remove_directory "${CMAKE_TEMP_HEADER_FOLDER}"
   COMMENT
     "Generating hourglass API bindings and implementation stub"
   DEPENDS
